File: zadanie1/haskell/silnia.hs
silnia i =
  if i<=0
  then 1
  else i*silnia(i-1)

silniaiteracyjnie i = product[1..i]  

main = do
  putStrLn "podaj liczbe z ktorej chcesz silnie policzyc"
  liczba <- getLine
  putStrLn "silnia rekurencyjnie"
  print $ silnia(read liczba :: Int)
  putStrLn "silnia iteracyjnie"
  print $ silniaiteracyjnie(read liczba :: Int)
  

File: zadanie1/c/silnia.c
#include <stdio.h>
int silnia(int number)
{
  int result = 1;
  for(number;number>1;number--)
    result*=number;
  return result;

}

int silniarek(int number)
{
  if(number>0)
    return number*silniarek(number-1);
  return 1;
}

int main()
{
  
  printf("podaj liczbe na ktorej n! chcesz policzyc\n");
  unsigned int number;
  scanf("%ud",&number);
  printf("Silnia interacyjnie:  %d\nsilnia rekurencyjnie: %d\n",silnia(number),silniarek(number));
    
  
}
